runtime/bcutil/defineclass.c

```diff
@@ -549,7 +549,11 @@ internalLoadROMClass(J9VMThread * vmThread, J9LoadROMClassData *loadData, J9Tran
 
 	/* Determine allowed class file version */
 #ifdef J9VM_OPT_SIDECAR
-	translationFlags |= BCT_JavaMajorVersionShifted(JAVA_SPEC_VERSION);
+	{
+		/* majorVer is to accommodate JDK8 zOS 64bit */
```

Comment:
This comment deserves to be improved - explaining that it's a workaround for an apparent compiler problem.

The better solution (assuming it works) is to change `j9nonbuilder.h`:
```
#define BCT_JavaMajorVersionShifted(java_version) ((44 + (U_32)(java_version)) << BCT_MajorClassFileVersionMaskShift)
```
as all the uses appear to be in unsigned contexts.

Reply:
Updated the comment.

Modifying the cast to `U_32` doesn't help, change it anyway as per the comment _all the uses appear to be in unsigned contexts_

Please have another look.